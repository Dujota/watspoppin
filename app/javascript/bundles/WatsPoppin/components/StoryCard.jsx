import React from 'react';
import PropTypes from 'prop-types';

export default class StoryCard extends React.Component {
  static propTypes = {
    story: PropTypes.object.isRequired
  };

  render() {
    const { story } = this.props;
    const image_url = story.image.thumb.url;

    return (
      <figure>
        <img
          className={image_url ? 'user_img' : 'default_img'}
          src={image_url ? image_url : '/twitter-logo.svg'}
          alt={story.content}
        />
        <figcaption>{story.content}</figcaption>
      </figure>
    );
  }
}
