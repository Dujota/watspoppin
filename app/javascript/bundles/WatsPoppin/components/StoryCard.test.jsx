import StoryCard from './StoryCard';

const story = {
  id: 1,
  content: 'Rapido corren los carros',
  image: {
    thumb: {
      url:
        'https://s7d2.scene7.com/is/image/PetSmart/PB1201_STORY_CARO-Authority-HealthyOutside-DOG-20160818?$PB1201$'
    }
  }
};

it('Renders a figure for the story clicked', () => {
  const storyCard = render(<StoryCard story={story} />);

  expect(storyCard).toMatchSnapshot();
});
