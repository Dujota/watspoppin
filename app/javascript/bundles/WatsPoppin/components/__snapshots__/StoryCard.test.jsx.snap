// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a figure for the story clicked 1`] = `
<figure>
  <img
    alt="Rapido corren los carros"
    class="user_img"
    src="https://s7d2.scene7.com/is/image/PetSmart/PB1201_STORY_CARO-Authority-HealthyOutside-DOG-20160818?$PB1201$"
  />
  <span
    class="Linkify"
  >
    <figcaption>
      Rapido corren los carros
    </figcaption>
  </span>
</figure>
`;
